app/models/host_status/configuration_status.rb

```diff
@@ -109,10 +109,11 @@ def self.bit_mask(config_status)
     end
 
     def status_link
-      return nil if last_report.nil?
-      return nil unless User.current.can?(:view_config_reports, last_report)
+      return @config_status_link if defined?(@config_status_link)
+      return @config_status_link = nil if last_report.nil?
+      return @config_status_link = nil unless User.current.can?(:view_config_reports, last_report, false)
 
-      last_report && Rails.application.routes.url_helpers.config_report_path(last_report)
+      @config_status_link = last_report && Rails.application.routes.url_helpers.config_report_path(last_report)
     end
```

Comment:
View code in a model. Foreman in 2021 :-)

I like the solution.

Reply:
this code should definitely move to a helper :)